[Tests] Update to use React Native Testing Library Jest Matchers and add accessibility tests

narin opened the pull request:
<!-- PR title naming convention:
'[Issue type] Brief summary of issue suitable for changelog or copy/paste issue title',
where Issue type = bug, feature, spike, CU (code upkeep), etc.-->

<!-- Preferred branch naming convention:
'[Issue type]/[Issue #]-[Your name]-[Summary of issue]',
where Issue type = bug, feature, spike, CU (code upkeep), etc.-->

## Description of Change
<!-- Describe the change and context with which it was made beyond ACs unless straightforward.
Consider:
 - What is relevant to code reviewer(s) and not in the ticket?
 - What context may be relevant to a future dev or you in 6 months about this PR?
 - Did the course of work lead to notable dead ends? If so, why didn't they pan out?
 - Did the change add new dependencies? Why?
 - Were there important sources to link? Examples: an open bug with a dependency project, an article of someone else solving the same problem that was partially or wholly copied, external documentation relevant to solution
 -->
Adds React Native Testing Library's Jest matchers as per [setup instructions](https://callstack.github.io/react-native-testing-library/docs/jest-matchers#setup). These matchers provide us with helpful matchers, make tests more readable, and also introduces RNTL specific linting to help us adhere to best practices. This required some updates to our jest and TypeScript setup, as well as changes to all of our tests. Most notable changes:

1. Usage of `screen` for queries instead of assigning the result of the `render` to a variable like we've been doing with `component` (enforced by [testing-library/prefer-screen-queries](https://github.com/testing-library/eslint-plugin-testing-library/blob/main/docs/rules/prefer-screen-queries.md))
2. `render` is not allowed in `beforeEach()` (enforced by [testing-library/no-render-in-lifecycle](https://github.com/testing-library/eslint-plugin-testing-library/blob/main/docs/rules/no-render-in-lifecycle.md)). Added `render()`'s to each test.
3. Replace `toBeTruthy` and `toBeNull` tests with more readable `toBeOnTheScreen` and `not.toBeOnTheScreen`
4. Replace existing accessibility tests in Link with accessibility queries such as `getByLabel` and `getByHintText` or use `hasAccessibilityValue` matcher
5. Add accessibility tests to SegmentedControl, Button, and Alert
6. Replace existing style tests with `toHaveStyle` matcher
7. Replace any color tokens with hex values until color tokens are finalized
8. Wrap Link component in ComponentWrapper. Translated `aria-valuetext` wasn't coming through for usage with `hasAccessibilityValue`


#### Testing Packages
<!-- List or range of alpha/beta packages published in association with this PR, if any -->


### Screenshots/Video
<!-- Add screenshots or video as needed; before/after recommended if appropriate. 
Convenience side-by-side formatting:
Before/after: <img src="" width="49%" />&nbsp;&nbsp;<img src="" width="49%" />
Accordion before/after: <details><summary>Before/after</summary><img src="" width="49%" />&nbsp;&nbsp;<img src="" width="49%" /></details>
-->
![Screenshot 2024-05-02 at 12 21 22 PM](https://github.com/department-of-veterans-affairs/va-mobile-library/assets/786704/4a74be9d-e1f6-4416-94d9-dd9ff970c364)



## Testing
<!-- Describe testing conducted to validate changes.
Consider highlighting:
- What testing was not explicitly done and may be relevant for QA? 
- Edge cases validated
- Special situations that could not be tested
- Any testing performed in a consuming app -->

N/A – Updates to tests only

## PR Checklist
Code reviewer validation:
- General
	- [x] PR is linked to ticket(s)
	- [x] PR has `changelog` label applied if it's to be included in the changelog
	- [x] Acceptance criteria: 
		- All satisfied _or_
		- Documented reason for not being performed _or_
		- Split to separate ticket and ticket is linked by relevant AC(s)
	- [x] Above PR sections adequately filled out
	- [x] If any breaking changes, in accordance with the [pre-1.0.0 versioning guidelines](https://github.com/department-of-veterans-affairs/va-mobile-library#versioning-policy): a CU ticket has been created for the VA Mobile App detailing necessary adjustments with the package version that will be published by this ticket
- Code
	- [x] Tests are included if appropriate (or split to separate ticket)
	- [x] New functions have proper TSDoc annotations

## Publish
<!-- Most changes entail a version increment; section can be removed for PRs exclusively within non-ship-relevant files (e.g. unit tests, Storybook stories) -->
If changes warrant a new version [per the versioning guidelines](https://github.com/department-of-veterans-affairs/va-mobile-library#versioning-policy) and the PR is approved and ready to merge:
- [x] Merge `main` into branch
- [ ] Merge branch to `main`
- [ ] Verify that [Check Component Integrations](https://github.com/department-of-veterans-affairs/va-mobile-library/actions/workflows/check-component-integrations.yml) workflow ran successfully
- [ ] [Publish new version](https://github.com/department-of-veterans-affairs/va-mobile-library/actions/workflows/publish.yml) after merging

TimRoe commented on packages/components/tsconfig.json:
```diff
@@ -1,5 +1,7 @@
 {
   "compilerOptions": {
+    "module": "NodeNext",
+    "moduleResolution": "NodeNext",
```
These are already present (commented out) down in the "Modules" section if we were continuing to roll with the structured form of key/values.

TimRoe commented on packages/components/src/components/SegmentedControl/SegmentedControl.test.tsx:
```diff
@@ -27,26 +20,31 @@ describe('SegmentedControl', () => {
   })
 
   const labels = ['segment 0', 'segment 1']
+  const a11yLabels = ['segment 0 a11y label', 'segment 1 a11y label']
```
Wonder if this should only override one or the other to validate the fallback case of using `labels` when there's a falsy value in the array.

narin commented on packages/components/src/components/SegmentedControl/SegmentedControl.test.tsx:
```diff
@@ -56,65 +54,94 @@ describe('SegmentedControl', () => {
     }
   }
 
-  beforeEach(() => {
```
`beforeEach` was calling `initializeTestInstance` which calls `render` inside of it which goes against the [testing-library/no-render-in-lifecycle](https://github.com/testing-library/eslint-plugin-testing-library/blob/main/docs/rules/no-render-in-lifecycle.md) linter rule

TimRoe commented on packages/components/src/components/Alert/Alert.test.tsx:
```diff
@@ -18,292 +18,326 @@ jest.mock('react-native/Libraries/Utilities/useColorScheme', () => {
-        expect(component.queryByText('Secondary test button')).toBeNull()
-      })
+    it('should render only the header after being collapsed', () => {
+      render(
+        <Alert
+          variant="error"
+          expandable
+          initializeExpanded={true}
+          {...commonProps}
+        />,
+      )
```
Did this test need to render everything again or would it still be on screen in collapsed form?  Curious since the render for the prior test is outside the test and doesn't change anything. Maybe this is also bumping up against the linter though.

narin replied:
Yes, the test would fail with `'render' method has not been called` if there hasn't been a "fresh" render called prior to the test. I suspect the previous test didn't fail since render is called right before, even if it wasn't within the it(). Moved the render to be within.

TimRoe commented on packages/components/src/components/Link/Link.test.tsx:
```diff
@@ -43,57 +42,53 @@ describe('Link', () => {
   const defaultProps: LinkProps = {
```
Maybe worth renaming `commonProps` for consistency with other tests?

narin replied:
Renamed

TimRoe commented on packages/components/src/components/Link/Link.test.tsx:
```diff
@@ -424,28 +411,30 @@ describe('Link', () => {
+      expect(screen.getByHintText('a11yHint override')).toBeOnTheScreen()
     })
 
     it('includes a11yValue', async () => {
```
Mentioned in our 1:1 but probably not very clearly, I think we want a second a11yValue test for poking at sending a custom one. 
> /** Optional a11yValue for "[position #] of [list total #]" or a custom value descriptive string */
  a11yValue?: a11yValue | string

So basically a test rendering with defaultProps + a11yValue sent as a string (AKA they have a custom formatted string description and didn't want to leverage our translation based on indexes). It was done this way as [accessibilityValue is rather flexible in use conditions](https://reactnative.dev/docs/accessibility#accessibilityvalue).

TimRoe reviewed (changes requested):
Looks mostly good. Left a couple question/suggestion comments, but main primarily requesting changes for:
1. Adding additional Link a11y test to validate sending a custom string for `a11yValue`
2. Adjusting the Segmented Control a11yLabels const to `[undefined, 'segment 1 a11y label']` and validating that the a11yLabel for the first segment falls back to `segment 0` to validate the fallback behavior of using the label itself without override present for that segment's index

narin commented:
- Added additional Link a11yValue test for custom string
- Updated SegmentedControl a11yLabel test to account for fallback value
- Renamed `defaultProps` to `commonProps` in Link
